scopes/ui-foundation/tree/folder-tree-node/folder-tree-node.tsx

```diff
@@ -1,18 +1,26 @@
 import { Icon } from '@teambit/design.elements.icon';
 import classNames from 'classnames';
 import React, { ReactNode, useState, useEffect } from 'react';
-import AnimateHeight from 'react-animate-height';
 
-import { TreeLayer, TreeNodeProps } from '@teambit/base-ui.graph.tree.recursive-tree';
-import { indentClass, indentStyle } from '@teambit/base-ui.graph.tree.indent';
+import { TreeNodeProps } from '@teambit/base-ui.graph.tree.recursive-tree';
+import { indentClass } from '@teambit/base-ui.graph.tree.indent';
+import { CollapsableTreeNode } from '@teambit/base-ui.graph.tree.collapsable-tree-node';
+import type { FolderPayload } from '@teambit/base-ui.graph.tree.collapsable-tree-node';
 import styles from './folder-tree-node.module.scss';
 
-export type FolderTreeNodeProps = {
-} & TreeNodeProps<FolderPayload>;
+export type FolderTreeNodeProps = {} & TreeNodeProps<FolderPayload>;
 
-export type FolderPayload = {
-  icon?: string | ReactNode;
-  open?: boolean;
```

Comment:
this is actually the right place - `FolderTreeNode` decides what to do with the `icon` and `open`. 
DocsTreeFolder can decide to want `{ icon, open, defaultLink: string }`, etc.

collapsable-tree-node can stay generic to any tree node.